src/views/InvolvementProfile/components/Membership/components/MemberList/index.js

```diff
@@ -93,12 +93,38 @@ const MemberList = ({
       style={headerStyle}
     />
   );
+  /// function that re-arrange names alpabetically and also by Advisor, Leader, members and Guest
+  // by Hierarchy.
+  function sortByParticipation(memberArr) {
```

Comment:
You have kind of imitated sorting behavior by splitting the data into arrays and the rejoining it in a new order. That certainly works, but it's highly inefficient and also more complicated than necessary. We already have a sorting function, `compareByLastThenFirst` (lines 36-50) that you could extend to sort by participation first, and then name. That would be more efficient and more understandable I think.

Reply:
Here's what I came up with after a couple minutes of thinking about it:
```JavaScript
const compareByParticipationThenLastThenFirst = (a, b) => {
    if (a.ParticipationDescription !== b.ParticipationDescription) {
      if (a.ParticipationDescription === 'Advisor') return 1;
      if (b.ParticipationDescription === 'Advisor') return -1;
      if (a.ParticipationDescription === 'Leader') return 1;
      // Handle other cases
    } else {
      if (a.LastName.toUpperCase() < b.LastName.toUpperCase()) return -1;

      if (a.LastName.toUpperCase() > b.LastName.toUpperCase()) return 1;

      if (a.FirstName.toUpperCase() < b.FirstName.toUpperCase()) return -1;

      if (a.FirstName.toUpperCase() > b.FirstName.toUpperCase()) return 1;
    }

    return 0;
  };
```

We know that if two memberships have the same ParticipationDetail, then we compare them by the other properties. If they don't, then we just need to find which one has the highest priority ParticipationDetail. e.g. if `a` is an `Advisor` and we know `b` is NOT and `Advisor`, than `a` is higher sort order.

Reply:
Alternatively, you could define an Enum that encodes the relative hierarchy:
```javascript
const ParticipatioinLevels = {
  Guest: 0,
  Member: 1,
  Leader:2,
  Advisor: 3
}
```
 and use that for comparison:
```javascript
    if (ParticipationLevels[a.ParticipationDescription] > ParticipationLevels[b.ParticipationDescription]) return 1;
    if (ParticipationLevels[a.ParticipationDescription] < ParticipationLevels[b.ParticipationDescription]) return -1;
```

Reply:
I tried the first one that you suggestion which work fine and it looks a lot cleaner than what i wrote before, and makes more sense and easy to read.